dockstore-webservice/src/main/java/io/dockstore/webservice/resources/WorkflowResource.java

```diff
@@ -568,6 +568,9 @@ private void updateInfo(Workflow oldWorkflow, Workflow newWorkflow) {
             oldWorkflow.setTopicSelection(newWorkflow.getTopicSelection());
         }
 
+        // Update DOI selection
```

Comment:
Could be beyond the scope of this PR, but I've been confused by this method and the endpoint calling it before -- it only updates a subset of the possible fields, of which you've now added one. I thought it would update the entire workflow...

TLDR; updating the comment on line 42, even to say something like "updates some of the workflow", so you'll know to look at the code for what's actually updated, would be nice. But not necessary.

Reply:
Yeah, I'm not sure what the preferred way to update a field is (using the existing updateWorkflow endpoint vs creating a new endpoint) and I went with the existing behaviour using the updateWorkflow endpoint

Reply:
Some thoughts that roughly correspond with where I'm at
https://softwareengineering.stackexchange.com/questions/208271/how-should-a-rest-api-handle-put-requests-to-partially-modifiable-resources

e.g. 
> A nice thing to do if you choose this option would be to send back a 200 (OK) with the actual updated entity in the response, so that clients can clearly see that the read-only fields were not updated.

Don't feel strongly about new endpoint vs updating this one, but just putting it out there that this kind of ambiguity in PUT calls is probably not that uncommon.

Reply:
Some thoughts, no need to make changes:
1. One source of confusion for me is that this method's argument names are ambiguous.  Are we updating the "old" or the "new" workflow?  "src" and "dst" would be more clear.
2. When there's a complicated object that we want to update piecemeal, a verb-based API can work well.  PUTting the  whole thing and then attempting to suss what/how to update and how to do it doesn't scale from a robustness POV.